<launch>

<arg name="urdf_file" default="$(find xacro)/xacro.py '$(find watermellon)/robots/kobuki/kobuki_hexagons_asus_xtion_pro.urdf.xacro'" />
  <param name="robot_description" command="$(arg urdf_file)" />
  
  <node pkg="nodelet" type="nodelet" name="mobile_base_nodelet_manager" args="manager"/>
  <node pkg="nodelet" type="nodelet" name="mobile_base" args="load kobuki_node/KobukiNodelet mobile_base_nodelet_manager">
    <rosparam file="$(find kobuki_node)/param/base.yaml" command="load"/>
    <!--<remap from="mobile_base/odom" to="odom"/>-->
    <!-- Don't do this - force applications to use a velocity mux for redirection
      <remap from="mobile_base/commands/velocity" to="cmd_vel"/>
    -->
    <remap from="mobile_base/enable" to="enable"/>
    <remap from="mobile_base/disable" to="disable"/>
    <remap from="mobile_base/joint_states" to="joint_states"/>
  </node>

  <node pkg="diagnostic_aggregator" type="aggregator_node" name="diagnostic_aggregator" >
    <rosparam command="load" file="$(find turtlebot_bringup)/param/kobuki/diagnostics.yaml" />
  </node>
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="30.0" />
  </node>


<node pkg="watermellon" type="wm_map_server_node" name="wm_map_server" args="'$(find watermellon)/maps/urjc/lab.pcd'">
		<param name="frame_id" type="string" value="/map" />
		<param name="mapping" type="boolean" value="false" />
		<param name="pointcloud_min_z" type="double" value="-0.5" /> 
		<param name="pointcloud_max_z" type="double" value="2.0" />
		<param name="pointcloud_max_x" type="double" value="4.0" />
		<param name="pointcloud_max_t_xy" type="double" value="0.3" />
		<param name="pointcloud_max_t_xz" type="double" value="0.4" />
		
		<remap from="cloud_in" to="/camera/depth_registered/points" />
	</node>


 <node pkg="watermellon" type="wm_localization_node" name="wm_localization" output="screen">
  		<param name="frame_id" type="string" value="map" />
 		<param name="pointcloud_min_z" type="double" value="-0.5" /> 
		<param name="pointcloud_max_z" type="double" value="2.0" />
		<param name="pointcloud_max_x" type="double" value="4.0" />
		<param name="pointcloud_max_t_xy" type="double" value="0.3" />
		<param name="pointcloud_max_t_xz" type="double" value="0.4" />
		<param name="initial_pose_x" type="double" value="0.0" />
 		<param name="initial_pose_y" type="double" value="0.0" />
 		<param name="initial_pose_z" type="double" value="0.0" />
  		<remap from="cloud_in" to="/camera/depth_registered/points" />
  	</node> 


	<node pkg="rviz" type="rviz" output="screen"
		name="rviz"
		args="-d $(find watermellon)/rviz/wm_localization.rviz"/>

</launch>
